include/bluetooth/services/ots.h

```diff
@@ -0,0 +1,600 @@
+enum {
+	/** Bit 0 OLCP Go To Op Code Supported */
+	BT_GATT_OTS_OLCP_FEAT_GO_TO      = 0,
+
+	/** Bit 1 OLCP Order Op Code Supported */
+	BT_GATT_OTS_OLCP_FEAT_ORDER      = 1,
+
+	/** Bit 2 OLCP Request Number of Objects Op Code Supported */
+	BT_GATT_OTS_OLCP_FEAT_NUM_REQ    = 2,
+
+	/** Bit 3 OLCP Clear Marking Op Code Supported*/
+	BT_GATT_OTS_OLCP_FEAT_CLEAR      = 3,
+};
+
+/** @brief Set @ref BT_GATT_OTS_OLCP_FEAT_GO_TO feature.
+ *
+ *  @param feat OTS features
+ */
+#define BT_GATT_OTS_OLCP_SET_FEAT_GO_TO(feat) \
+	WRITE_BIT(feat, BT_GATT_OTS_OLCP_FEAT_GO_TO, 1)
+
+/** @brief Set @ref BT_GATT_OTS_OLCP_FEAT_ORDER feature.
+ *
+ *  @param feat OTS features
+ */
+#define BT_GATT_OTS_OLCP_SET_FEAT_ORDER(feat) \
+	WRITE_BIT(feat, BT_GATT_OTS_OLCP_FEAT_ORDER, 1)
+
+/** @brief Set @ref BT_GATT_OTS_OLCP_FEAT_NUM_REQ feature.
+ *
+ *  @param feat OTS features
+ */
+#define BT_GATT_OTS_OLCP_SET_FEAT_NUM_REQ(feat) \
+	WRITE_BIT(feat, BT_GATT_OTS_OLCP_FEAT_NUM_REQ, 1)
+
+/** @brief Set @ref BT_GATT_OTS_OLCP_FEAT_CLEAR feature.
+ *
+ *  @param feat OTS features
+ */
+#define BT_GATT_OTS_OLCP_SET_FEAT_CLEAR(feat) \
+	WRITE_BIT(feat, BT_GATT_OTS_OLCP_FEAT_CLEAR, 1)
+
+/** @brief Get @ref BT_GATT_OTS_OLCP_GET_FEAT_GO_TO feature.
+ *
+ *  @param feat OTS features
+ */
+#define BT_GATT_OTS_OLCP_GET_FEAT_GO_TO(feat) \
+	((feat) & BIT(BT_GATT_OTS_OLCP_FEAT_GO_TO))
+
+/** @brief Get @ref BT_GATT_OTS_OLCP_GET_FEAT_ORDER feature.
+ *
+ *  @param feat OTS features
+ */
+#define BT_GATT_OTS_OLCP_GET_FEAT_ORDER(feat) \
+	((feat) & BIT(BT_GATT_OTS_OLCP_FEAT_ORDER))
+
+/** @brief Get @ref BT_GATT_OTS_OLCP_GET_FEAT_NUM_REQ feature.
+ *
+ *  @param feat OTS features
+ */
+#define BT_GATT_OTS_OLCP_GET_FEAT_NUM_REQ(feat) \
+	((feat) & BIT(BT_GATT_OTS_OLCP_FEAT_NUM_REQ))
+
+/** @brief Get @ref BT_GATT_OTS_OLCP_GET_FEAT_CLEAR feature.
+ *
+ *  @param feat OTS features
+ */
+#define BT_GATT_OTS_OLCP_GET_FEAT_CLEAR(feat) \
+	((feat) & BIT(BT_GATT_OTS_OLCP_FEAT_CLEAR))
+
+/**@brief Features of the OTS. */
+struct bt_gatt_ots_feat {
+	/* OACP Features */
+	uint32_t oacp;
+
+	/* OLCP Features */
+	uint32_t olcp;
+};
+
+/** @brief Opaque OTS instance. */
+struct bt_gatt_ots;
+
+/** @brief OTS server callback structure. */
+struct bt_gatt_ots_cb {
+	/** Object created callback
+	 *
+	 *  This callback is called whenever a new object is created.
+	 *  Application can reject this request by returning an error
+	 *  when it does not have necessary resources to hold this new
+	 *  object.
+	 *
+	 *  @param ots  OTS Server instance
+	 *  @param conn The connection that is requesting object creation or
+	 *              NULL if object is created by bt_gatt_ots_object_add()
+	 *  @param id   Object ID
+	 *  @param init Object initialization metadata
+	 *
+	 *  @return 0 in case of success or negative value in case of error.
+	 *  Possible return values:
+	 *  -ENOMEM if no available space for new object.
+	 */
+	int (*obj_created)(struct bt_gatt_ots *ots, struct bt_conn *conn,
```

Comment:
There is a control point for the client to create objects on the server. I guess the API is here to allow that, while also be used when the server creates an object, so that there's a common object create path.

Reply:
Yes, this is correct. @Thalley, thank you for explaining it.

Currently, this server implementation does not support operations for creating and deleting objects by the client, but I wanted to have uniform way of notifying application about created objects (regardless if the operation was triggered locally by the Server or remotely by the Client).